cpp/serve/engine_state.h

```diff
@@ -89,17 +89,20 @@ class EngineStateObj : public Object {
   std::vector<Request> running_queue;
   /*! \brief The requests that have not started for process yet. */
   std::vector<Request> waiting_queue;
-  /*! \brief The states of all requests. */
-  std::unordered_map<String, RequestState> request_states;
+  /*!
+   * \brief The states of all requests.
+   * One request may have more than one state when `n` in generation config is more than 1.
+   */
+  std::unordered_map<String, std::vector<RequestState>> request_states;
```

Comment:
Consider have a data structure for `vector<RequestState>`

// docs here
struct RequestStateEntry {
};

// docs here
struct RequestState {
    std::vector<RequestStateEntry> entries;
};

Reply:
https://github.com/mlc-ai/mlc-llm/blob/98f3c915b86e0e3d2bdfeb8ad0ee375a27c55838/cpp/serve/request_state.h#L210-L221